### PR TITLE
Files build log hook fixes

### PR DESCRIPTION
**Pull request by FrNecas**
- handle %global and %if keywords correctly
- follow Fedora Packaging Guidelines when adding manpages

Resolves: #629

### CODE REVIEW

File: rebasehelper/plugins/build_log_hooks/files.py [hunk 1]
```diff
@@ -191,6 +191,17 @@ def _get_best_matching_files_section(cls, rebase_spec_file, file):
...
         return best_match_section or rebase_spec_file.get_main_files_section()
 
+    @classmethod
+    def _sanitize_path(cls, path):
+        """Changes the path to follow Fedora Packaging Guidelines."""
+        if path.startswith('%{_mandir}'):
+            manpage_start = path.rfind('/')
+            # find 2nd occurrence of '.', replace everything after it
+            # with * wildcard.
+            extension_start = path[manpage_start:].index('.', path[manpage_start:].index('.') + 1)
```
Comment (nforro) [inline rebasehelper/plugins/build_log_hooks/files.py]: This won't work with uncompressed man pages.

File: rebasehelper/plugins/build_log_hooks/files.py [hunk 2]
```diff
@@ -191,6 +191,17 @@ def _get_best_matching_files_section(cls, rebase_spec_file, file):
 
         return best_match_section or rebase_spec_file.get_main_files_section()
 
+    @classmethod
+    def _sanitize_path(cls, path):
+        """Changes the path to follow Fedora Packaging Guidelines."""
+        if path.startswith('%{_mandir}'):
+            manpage_start = path.rfind('/')
```
Comment (nforro) [inline rebasehelper/plugins/build_log_hooks/files.py]: What's wrong with `os.path.basename()`? Or `os.path.split()`, as you need the ret of the path.

File: rebasehelper/plugins/build_log_hooks/files.py [hunk 3]
```diff
@@ -191,6 +191,17 @@ def _get_best_matching_files_section(cls, rebase_spec_file, file):
...
+            # find 2nd occurrence of '.', replace everything after it
+            # with * wildcard.
+            extension_start = path[manpage_start:].index('.', path[manpage_start:].index('.') + 1)
+            path = path[:manpage_start+extension_start] + '*'
```
Comment (nforro) [inline rebasehelper/plugins/build_log_hooks/files.py]: Something like this should be sufficient:

```python
dir, name = os.path.split(path)
return os.path.join(dir, name.split('.')[:2] + '*')
```

Comment (FrNecas) [reply rebasehelper/plugins/build_log_hooks/files.py]: Thanks. Would you keep the separate function for doing this? There could be more path modifications that we are missing.

Comment (nforro) [reply rebasehelper/plugins/build_log_hooks/files.py]: Sure, why not.

**Pull request merged**